frontend/src/metabase/querying/analytics.ts

```diff
@@ -0,0 +1,40 @@
+import { trackSchemaEvent } from "metabase/lib/analytics";
+import * as Lib from "metabase-lib";
+import type Question from "metabase-lib/v1/Question";
+
+export const trackColumnExtractViaHeader = (
+  query: Lib.Query,
+  tag: string,
+  question?: Question,
+) => {
+  trackSchemaEvent("question", "1-0-4", {
+    event: "column_extract_via_column_header",
+    custom_expressions_used: expressionsUsedBy(tag),
+    database_id: Lib.databaseID(query),
+    question_id: question?.id() ?? 0,
+  });
+};
+
+function expressionsUsedBy(tag: string) {
+  switch (tag) {
```

Comment:
This is a bit fragile: when we add more extractions, we need to add the expressions used here too.

But AFAIC there is no easy way to get the expressions introduced by an extraction from the extraction itself.

Reply:
We could make `ColumnExtractionInfo.tag` to be a enum and not just a string. We could also define a mapping type used here in a way to make it fail if there is a new tag that is not supported here. Like:
```
Record<ColumnExtractionTag, string>
```

Another way could to be to make analytic events match the expression tag either by updating the tag or the analytics spec.

Reply:
Ok I've done exactly that, please have another look.